runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/Presigner.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package aws.smithy.kotlin.runtime.auth.awssigning
+
+import aws.smithy.kotlin.runtime.auth.awscredentials.CredentialsProvider
+import aws.smithy.kotlin.runtime.http.*
+import aws.smithy.kotlin.runtime.http.endpoints.Endpoint
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+import aws.smithy.kotlin.runtime.util.InternalApi
+import kotlin.time.Duration
+
+data class SigningContext(val service: String?, val region: String?)
+data class SigningContextualizedEndpoint(val endpoint: Endpoint, val context: SigningContext?)
+typealias SigningEndpointProvider = suspend (SigningContext) -> SigningContextualizedEndpoint
+
+/**
+ * The service configuration details for a presigned request
+ *
+ * @property region The AWS region to which the request is going
+ * @property signingName The signing name used to sign the request
+ * @property serviceId the service id used to sign the request
+ * @property endpointProvider Resolves the contextualized endpoint to determine where the request should be sent
+ * @property credentialsProvider Resolves credentials to sign the request with
+ * @property useDoubleUriEncode Determines if presigner should double encode Uri
+ * @property normalizeUriPath Determines if presigned URI path will be normalized
+ */
+interface ServicePresignConfig {
+    val signer: AwsSigner
+    val region: String
+    val signingName: String
+    val serviceId: String
+    val endpointProvider: SigningEndpointProvider
+    val credentialsProvider: CredentialsProvider
+    val useDoubleUriEncode: Boolean
+    val normalizeUriPath: Boolean
+}
+
+/**
+ * Where the signature is placed in the presigned request
+ * @property HEADER
+ * @property QUERY_STRING
+ */
+enum class SigningLocation {
```

Comment:
nit 

This name isn't specific to presigners, perhaps `PresignSignatureLocation`?